framework/src/modules/interoperability/base_interoperability_method.ts

```diff
@@ -103,6 +103,10 @@ export abstract class BaseInteroperabilityMethod<
 			updatedChainID = mainchainID;
 		}
 
+		if (chainID.equals(ownChainAccount.chainID)) {
```

Comment:
I would run this check before reading from the state (so before the `hasChainAccount` call)